addons/mail/static/src/messaging/component/activity_mark_done_popover/activity_mark_done_popover_tests.js

```diff
@@ -0,0 +1,280 @@
+
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover',
+        "Popover component should be present"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_feedback',
+        "Popover component should contain the feedback textarea"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_buttons',
+        "Popover component should contain the action buttons"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_doneScheduleNextButton',
+        "Popover component should contain the done & schedule next button"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_doneButton',
+        "Popover component should contain the done button"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_discardButton',
+        "Popover component should contain the discard button"
+    );
+});
+
+QUnit.test('activity with force next mark done popover simplest layout', async function (assert) {
+    assert.expect(6);
+
+    await this.start();
+    const activity = this.env.entities.Activity.create({
+        activity_category: 'not_upload_file',
+        can_write: true,
+        force_next: true,
+        id: 12,
+    });
+    await this.createActivityMarkDonePopoverComponent(activity);
+
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover',
+        "Popover component should be present"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_feedback',
+        "Popover component should contain the feedback textarea"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_buttons',
+        "Popover component should contain the action buttons"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_doneScheduleNextButton',
+        "Popover component should contain the done & schedule next button"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_ActivityMarkDonePopover_doneButton',
+        "Popover component should NOT contain the done button"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_ActivityMarkDonePopover_discardButton',
+        "Popover component should NOT contain the discard button"
+    );
+});
+
+QUnit.test('activity mark done popover click on discard', async function (assert) {
+    assert.expect(4);
+
+    await this.start();
+    const activity = this.env.entities.Activity.create({
+        activity_category: 'not_upload_file',
+        can_write: true,
+        id: 12,
+    });
+    await this.createActivityMarkDonePopoverComponent(activity);
+    document.addEventListener('o-discard-clicked', () => {
+        assert.step('event_triggered');
+    }, true);
```

Comment:
ugh! that's gross...
Also out of scope of PR, but would be nice popover state that depends on entity changes. Rough idea: causal relation activity/popover, so that when activity if deleted, popover also is deleted, and popover in component world is removed.

Reply:
I know but no idea to make it differently with the current situation... any better idea ?

Reply:
keep like this for the time being :)